mlflow/langchain/_langchain_autolog.py

```diff
@@ -96,15 +114,21 @@ def _invoke(self, *args, **kwargs):
             _log_optional_artifacts(config, run_id, result, self, func_name, *args, **kwargs)
     else:
         result = _invoke(self, *args, **kwargs)
+
+    if should_trace:
+        # Make sure all spans are flushed before finishing the inference. LangChain's on_xyz_end
+        # callbacks are not guaranteed to be invoked always, which results in leaking the active
+        # span context to the next inference call. Flushing the tracer ensures that all spans are
+        # finished and detached from the context.
+        tracer.flush()
```

Comment:
Should this be always called even when `_invoke` fails?

Reply:
Ah that's true, let me update the logic.